### PR TITLE
 Shotgun Pellet Number Reduction (and some minor QoL changes)

### PR DESCRIPTION
**Pull request by ChronicPwnage**
## Description

- Halved number of shotgun pellets (For buckshot, rubbershot, and improvised), doubled damage per pellet, and slightly tighened up spread.

<br>

- Increased number of logs recieved from cutting down dead trees to 5 instead of 3.
- Slightly increased crude barricade health (the barrier that goes on fences)
- You can now repair sandbags with sand
- Sandbags now visually smooth with material doors
- You can now repair iron and sandstone material doors with a welder (if you're not on harm intent)
- Plastic flaps are a little weaker (70 armor across the board instead of 80-100) - meaning they can now be destroyed with enough hits from melee weapons.
- You can now cut up bandanas to make cloth
- Fixed a few typos (like Big Leageus), incorrect text (first step of cutting a fence was "some more"), and missing text (like empty sandbag stacking)
- A few things now show visible messages to others when you do them (taking down sandbags, - barricading fences, etc)
- Increased the height of Tribal Torch sprite
- Increased the light provided by Tribal Torches
- Slightly increased the light provided by handheld torches
- Made tribal torches placable, requiring them to be snuffed out and uprooted to be moved around again (more work will be done on this later)

<br>

*EDIT:*
- Changed doubled shotgun damage values to be in line with their recently changed values
- Fixed empty sandbags disappearing once the stack reaches a threshold size.
## Motivation and Context

The shotgun changes are aimed at reducing lag while having a minimal impact on the functionality of shotguns. Local testing has shown some promising results.
The QoL changes are just fun little additions/changes/fixes that don't majorly affect gameplay but do improve the nuance and player experience of BD.
## How Has This Been Tested?

I have tested it on a local server with no problems.
## Changelog
:cl: ChronicPwnage
tweak: Halved number of shotgun pellets (For buckshot, rubbershot, and improvised), doubled damage per pellet, and slightly tighened up spread.
tweak: Increased number of logs recieved from cutting down dead trees to 5 instead of 3.
tweak: Slightly increased crude barricade health (the barrier that goes on fences)
add: You can now repair sandbags with sand
tweak: Sandbags now visually smooth with material doors
add: You can now repair iron and sandstone material doors with a welder (if you're not on harm intent)
tweak: Plastic flaps are a little weaker (70 armor across the board instead of 80-100) - meaning they can now be destroyed with enough hits from melee weapons.
add: You can now cut up bandanas to make cloth
bugfix: Fixed a few typos (like Big Leageus), incorrect text (first step of cutting a fence was "some more"), and missing text (like empty sandbag stacking)
tweak: A few things now show visible messages to others when you do them (taking down sandbags, - barricading fences, etc)
tweak: Increased the height of Tribal Torch sprite
tweak: Increased the light provided by Tribal Torches
tweak: Slightly increased the light provided by handheld torches
tweak: Made tribal torches placable, requiring them to be snuffed out and uprooted to be moved around again (more work will be done on this later)
bugfix: Fixed empty sandbags disappearing once the stack reaches a threshold size.
/:cl:

### CODE REVIEW

**Comment by ChronicPwnage** [comment]
I think the mood commit, which has already been added, somehow got mixed into this. It shouldn't be a problem code wise but sorry it's there, I'm still getting the hang of Github.

**Comment by improvedname** [comment]
Why the damage change on slugs?

File: code/modules/projectiles/projectile/bullets/shotgun.dm [hunk 1]
```diff
@@ -61,12 +60,12 @@
 
 /obj/item/projectile/bullet/pellet/shotgun_buckshot
 	name = "buckshot pellet"
-	damage = 7 //Up to 84 damage point blank
+	damage = 18 //Up to 108 damage point blank
```
Comment (improvedname) [inline code/modules/projectiles/projectile/bullets/shotgun.dm]: This needs to be 14 damage.

File: code/modules/projectiles/projectile/bullets/shotgun.dm [hunk 2]
```diff
@@ -61,12 +60,12 @@
...
 
 /obj/item/projectile/bullet/pellet/shotgun_rubbershot
 	name = "rubbershot pellet"
-	damage = 1
-	stamina = 7 //Does 84 pointplank and disregards armor reeeee
+	damage = 2
+	stamina = 20
```
Comment (improvedname) [inline code/modules/projectiles/projectile/bullets/shotgun.dm]: This needs to be 14 stamina damage.

File: code/modules/projectiles/projectile/bullets/shotgun.dm [hunk 3]
```diff
@@ -1,7 +1,6 @@
 /obj/item/projectile/bullet/shotgun_slug
 	name = "12g shotgun slug"
-	damage = 50
-	armour_penetration = 20
+	damage = 70
```
Comment (GhostEcho) [inline code/modules/projectiles/projectile/bullets/shotgun.dm]: This needs to be taken back as well to its previous damage

File: code/modules/projectiles/projectile/bullets/shotgun.dm [hunk 4]
```diff
@@ -79,7 +78,7 @@
 
 /obj/item/projectile/bullet/pellet/shotgun_improvised
 	tile_dropoff = 0.55		//Come on it does 6 damage don't be like that.
-	damage = 6
+	damage = 12
```
Comment (GhostEcho) [inline code/modules/projectiles/projectile/bullets/shotgun.dm]: It's not called improvised shell for nothing, but I do love me the effects on it

File: code/modules/projectiles/projectile/bullets/shotgun.dm (hunk 3 above)
Comment (improvedname) [inline code/modules/projectiles/projectile/bullets/shotgun.dm]: Revert this change

**Comment by rhicora** [comment]
Some balance changes here so I want to leave it up for comment for 24 hours. For those with a stake in balance: feel free to weigh in now.

**Comment by ChronicPwnage** [comment]
Okay so here's what happened:
I started editing code starting with the shotgun code, then realized it was behind and merged it with the up to date version, but I assumed shotgun code hadn't been touched in the time in between though it had, and as such I was halving old values. My bad, will fix, will make sure to merge FIRST in the future.

**Comment by UndeadNappist** [comment]
#260

**Comment by ChronicPwnage** [comment]
> 
> 
> #260

I can see that the shotgun changes are kind of a revert of this initial change, but my main reasons for proposing these changes (reduced pellets and spread) are:
- Lag reduction
- Ren asked me to

**Comment by UndeadNappist** [comment]
> I can see that the shotgun changes are kind of a revert of this initial change, but my main reasons for proposing these changes (reduced pellets and spread) are:
> 
> * Lag reduction
> * Ren asked me to

What's causing the lag and why? Why do bullets lag? Is it specifically shotguns or all guns?

image

**Comment by ChronicPwnage** [comment]
> 
> 
> > I can see that the shotgun changes are kind of a revert of this initial change, but my main reasons for proposing these changes (reduced pellets and spread) are:
> > 
> > * Lag reduction
> > * Ren asked me to
> 
> What's causing the lag and why? Why do bullets lag? Is it specifically shotguns or all guns?
> 
> image

I was just adding the Ren part to have them chime in on their reasons really. To me, it's about the lag.
As far as I can tell, it's the fact that there are 12 individual bullets being fired at once, all playing an individual hitsound, hit animation, and calculating damage on the player individually. I have only ever seen this type of "game freezes for a second or two" lag with shotguns, especially at point blank range. 

I do think it's a deeper issue. I have noticed similar lag on /tg/ when being hit by a lot of items at once due to a gravitational anomaly. There's something more to this for sure, but the fact of the matter is I'd rather have something that fixes a huge source of frustration for players now (I read complaints about shotgun lag in OOC daily) without majorly changing gameplay and THEN spend time looking for that deeper bug, rather than just keeping it the way it is until the deeper issue is found.

**Comment by Autismobirb** [comment]
My dudes and dudets, has anyone actually done extensive testing on the subject matter at hand.
And can actively cite there issues with some concrete proof.

**Comment by Autismobirb** [comment]
It's nice and all to say something is causing lag but I want more than just a "x said so" I want some good visual proof, code proof and a reason to why X does Y, and if this is over a micro-stutter on the client I'm going to literally shit a brick.

**Comment by UndeadNappist** [comment]
> As far as I can tell, it's the fact that there are 12 individual bullets being fired at once, all playing an individual hitsound, hit animation, and calculating damage on the player individually. I have only ever seen this type of "game freezes for a second or two" lag with shotguns, especially at point blank range.
> 
> I do think it's a deeper issue. I have noticed similar lag on /tg/ when being hit by a lot of items at once due to a gravitational anomaly. There's something more to this for sure,

>but the fact of the matter is I'd rather have something that fixes a huge source of frustration for players now (I read complaints about shotgun lag in OOC daily) without majorly changing gameplay and THEN spend time looking for that deeper bug, rather than just keeping it the way it is until the deeper issue is found.

Player complaints are valid and I can see that as a good reason to merge it immediately as a hotfix, then tackle the other issues... so long as the other reviewers get satisfied. The fact that multiple people are suffering this and you can go in and test it is objectively bad.

**Comment by JohnOxford** [comment]
As an issue that has existed for a while, even before i started coding here, generally fixes for said issues don't require testing. Assuming that the code at hand visually is functional in what it's trying to accomplish and that the damage changes for the shotgun rounds aren't merged along side of them.

"8:50 PM] JohnOxford: ill have a peek, i had to reinstall byond so its taking its time loading, i'll run through some bug fixes and see about moving the base
[8:51 PM] RenGusta ~ 自閉症: I prefer butfixes and optimisation over mapping, for nowm
[8:51 PM] RenGusta ~ 自閉症: Got this issue with shotguns, specifically buck and rubber shot
[8:52 PM] RenGusta ~ 自閉症: When you get hit by more than like 5 at once, it chokes you up, and in the case or point-blank, and multiple shots, can freeze you until you catch up, usually with dying.
[8:57 PM] JohnOxford: ill take a peek
"

that too

**Comment by ChronicPwnage** [comment]
So is this getting merged or does it require more discussion?

**Comment by improvedname** [comment]
just fill that changelog in i edited into your PR headbody

**Comment by ChronicPwnage** [comment]
> 
> 
> just fill that changelog in i edited into your PR headbody

done

**Comment by improvedname** [comment]
gucci gang
